analyzer/runtime/extract.go

```diff
@@ -727,6 +728,144 @@ func extractEvents(blockData *BlockData, relatedAccountAddresses map[apiTypes.Ad
 					}
 					return nil
 				},
+				ERC721Transfer: func(fromEthAddr []byte, toEthAddr []byte, tokenIDU256 []byte) error {
+					tokenID := &big.Int{}
+					tokenID.SetBytes(tokenIDU256)
+					fromZero := bytes.Equal(fromEthAddr, uncategorized.ZeroEthAddr)
+					toZero := bytes.Equal(toEthAddr, uncategorized.ZeroEthAddr)
+					if !fromZero {
+						fromAddr, err2 := registerRelatedEthAddress(blockData.AddressPreimages, relatedAccountAddresses, fromEthAddr)
+						if err2 != nil {
+							return fmt.Errorf("from: %w", err2)
+						}
+						eventData.RelatedAddresses[fromAddr] = true
+					}
+					if !toZero {
+						toAddr, err2 := registerRelatedEthAddress(blockData.AddressPreimages, relatedAccountAddresses, toEthAddr)
+						if err2 != nil {
+							return fmt.Errorf("to: %w", err2)
+						}
+						eventData.RelatedAddresses[toAddr] = true
+					}
+					// todo: reckon ownership
+					if _, ok := blockData.PossibleTokens[eventAddr]; !ok {
+						blockData.PossibleTokens[eventAddr] = &evm.EVMPossibleToken{}
+					}
+					// Mark as mutated if transfer is between zero address
+					// and nonzero address (either direction) and nonzero
+					// amount. These will change the total supply as mint/
+					// burn.
+					if fromZero != toZero && tokenID.Cmp(&big.Int{}) != 0 {
+						blockData.PossibleTokens[eventAddr].Mutated = true
+					}
+					eventData.EvmLogName = evmabi.ERC721.Events["Transfer"].Name
+					eventData.EvmLogSignature = ethCommon.BytesToHash(event.Topics[0])
+					eventData.EvmLogParams = []*apiTypes.EvmEventParam{
+						{
+							Name:    "from",
+							EvmType: "address",
+							Value:   ethCommon.BytesToAddress(fromEthAddr),
```

Comment:
Is this format compatible with what https://github.com/oasisprotocol/oasis-indexer/pull/421 (or rather, an analyzer that makes use of it) will eventually produce?
I guess even if it's not, we can leave it in and later change the parse; this is intended primarily for human consumption, and stability doesn't matter that much. But it would be nice if we don't have to change it.

Reply:
> Is this format compatible with what #421

ya, the go-ethereum library's own address type